motor/rule_applicator.go

```diff
@@ -111,6 +114,10 @@ func ApplyRulesToRuleSet(execution *RuleSetExecution) *RuleSetExecutionResult {
 	docConfig := datamodel.NewDocumentConfiguration()
 	//docConfig.SkipCircularReferenceCheck = true
 
+	if execution.IgnoreCircularArrayRef {
```

Comment:
This is great! My only request here is that we do the same for polymorphic refs as well, so both can be controlled by flags.

Reply:
Yesterday I could not make polymorphic test to pass with the example taken from the mentioned website.